core/cloudflow-operator/src/main/scala/cloudflow/operator/action/TopicActions.scala

```diff
@@ -123,71 +123,90 @@ object TopicActions {
-    createAction(namespace, labels, topicInfo)
+    createAction(newApp, namespace, labels, topicInfo)
   }
 
-  def maybeCreateActionFromAppConfigSecret(secretOption: Option[Secret], namespace: String, labels: CloudflowLabels, topic: TopicInfo) =
+  def maybeCreateActionFromAppConfigSecret(secretOption: Option[Secret],
+                                           newApp: CloudflowApplication.CR,
+                                           namespace: String,
+                                           labels: CloudflowLabels,
+                                           topic: TopicInfo) =
     for {
       secret <- secretOption
       config = ConfigInputChangeEvent.getConfigFromSecret(secret)
       kafkaConfig <- getKafkaConfig(config, topic)
       topicWithKafkaConfig = TopicInfo(Topic(id = topic.id, config = kafkaConfig))
       _ <- topicWithKafkaConfig.bootstrapServers
-    } yield createAction(namespace, labels, topicWithKafkaConfig)
+    } yield createAction(newApp, namespace, labels, topicWithKafkaConfig)
 
-  def createAction(appNamespace: String, labels: CloudflowLabels, topic: TopicInfo): CreateOrUpdateAction[ConfigMap] = {
-    val (bootstrapServers, partitions, replicas, brokerConfig) = (topic.bootstrapServers, topic.partitions, topic.replicationFactor) match {
-      case (Some(bootstrapServers), Some(partitions), Some(replicas)) => (bootstrapServers, partitions, replicas, topic.brokerConfig)
-      case _ =>
-        throw new Exception(
-          s"Default Kafka connection configuration was invalid for topic [${topic.name}]" +
-              topic.cluster.map(c => s", cluster [$c]").getOrElse("") +
-              ". Update installation of Cloudflow with Helm charts to include a default Kafka cluster configuration that contains defaults for 'bootstrapServers', 'partitions', and 'replicas'."
-        )
-    }
-    val configMap   = resource(appNamespace, topic, partitions, replicas, bootstrapServers, labels)
-    val adminClient = KafkaAdmins.getOrCreate(bootstrapServers, brokerConfig)
-
-    new CreateOrUpdateAction[ConfigMap](configMap, implicitly[Format[ConfigMap]], implicitly[ResourceDefinition[ConfigMap]], editor) {
-      override def execute(
-          client: KubernetesClient
-      )(implicit sys: ActorSystem, ec: ExecutionContext, lc: skuber.api.client.LoggingContext): Future[Action[ConfigMap]] =
-        super.execute(client).flatMap { resourceCreatedAction =>
-          createTopic().map(_ => resourceCreatedAction)
-        }
+  def createAction(newApp: CloudflowApplication.CR,
+                   appNamespace: String,
+                   labels: CloudflowLabels,
+                   topic: TopicInfo): Action[ObjectResource] =
+    (topic.bootstrapServers, topic.partitions, topic.replicationFactor) match {
+      case (Some(bootstrapServers), Some(partitions), Some(replicas)) =>
+        val brokerConfig = topic.brokerConfig
```

Comment:
Possibly move the good path into a method to make the `match` span over less code.

Reply:
done